kodiak.go

```diff
@@ -262,3 +273,25 @@ func (k *KodiakLPPriceProvider) getUnderlyingBalances(ctx context.Context) (*big
 		return reserves.Reserve0, reserves.Reserve1, err
 	}
 }
+
+// get contract byte code and SHA256 hash it to check the contract type against known values
+func checkContractType(ctx context.Context, client *ethclient.Client, address common.Address) ContractType {
+	bytecode, err := client.CodeAt(ctx, address, nil)
+	if err != nil {
+		panic(err)
+	}
+	byteCodeString := hex.EncodeToString(bytecode)
+
+	hasher := sha256.New()
+	hasher.Write([]byte(byteCodeString))
+	hash := hex.EncodeToString(hasher.Sum(nil))
+
+	switch hash {
+	case V2PoolByteCodeSHA256:
+		return ContractTypeV2Pool
+	case V3IslandByteCodeSHA256:
+		return ContractTypeV3Island
+	default:
+		return ContractTypeUnknown
```

Comment:
Do need to return `ContractTypeUnknown`, (you can also remove `ContractTypeUnknown`). Just return error here and forward it later.